### PR TITLE
DIRECTOR: Fix film loops with more than 8 cast members

### PR DESCRIPTION
**Pull request by moralrecordings**

### CODE REVIEW

File: engines/director/castmember.cpp [hunk 1]
```diff
@@ -986,8 +986,16 @@ Common::Array<Channel> *FilmLoopCastMember::getSubChannels(Common::Rect &bbox, C
...
 
 		_subchannels.push_back(chan);
-		_subchannels[_subchannels.size() - 1].replaceWidget();
+
+	}
+	// Initialise the widgets on all of the subchannels.
+	// This has to be done once the list has been constructed, otherwise
+	// the list grow operation will erase the widgets as they aren't
+	// part of the Channel assignment constructor.
+	for (Common::Array<Channel>::iterator iter = _subchannels.begin(); iter != _subchannels.end(); ++iter) {
```
Comment (sev-) [inline engines/director/castmember.cpp:995]: I recommend writing: `for (auto &iter : _subchannels)`

File: engines/director/transitions.cpp [hunk 2]
```diff
@@ -821,7 +829,9 @@ void Window::dissolvePatternsTrans(TransParams &t, Common::Rect &clipRect, Graph
 			break;
 		}
 
-		g_system->delayMillis(t.stepDuration);
+		uint32 endTime = g_system->getMillis(true);
```
Comment (sev-) [inline engines/director/transitions.cpp:832]: Why are you asking to skip Event Recorder here, e.g. `true` parameter? It should be used with `true` only within the event recorder itself

File: engines/director/transitions.cpp [hunk 3]
```diff
@@ -834,6 +844,7 @@ void Window::transMultiPass(TransParams &t, Common::Rect &clipRect, Graphics::Ma
 	Common::Array<Common::Rect> rects;
 
 	for (uint16 i = 1; i < t.steps; i++) {
+		uint32 startTime = g_system->getMillis(true);
```
Comment (sev-) [inline engines/director/transitions.cpp:847]: Same thing regarding `true` and in the instances below

**Comment by sev-** [comment]
Good thing. Have several notes. The PR title is a bit misleading, but this is absolutely fine.

Special thanks for the sound debug level.

File: engines/director/castmember.cpp (hunk 1 above)
Comment (moralrecordings) [reply engines/director/castmember.cpp]: Oh my word that is so much nicer! Glad to be rid of the C++ iterator dance.

File: engines/director/transitions.cpp (hunk 2 above)
Comment (moralrecordings) [reply engines/director/transitions.cpp]: Oops, fixed in all the spots.

**Comment by moralrecordings** [comment]
All fixed. I removed the commit for triggering sound playback before transitions; the actual behavior is more nuanced (e.g. script-initiated transitions don't stop the audio) and needs more investigation.

**Comment by sev-** [comment]
Thank you!

**Pull request merged**